processor/k8sattributesprocessor/internal/kube/client.go

```diff
@@ -313,6 +306,12 @@ func (c *WatchClient) extractPodAttributes(pod *api_v1.Pod) map[string]string {
 				if c.Rules.ReplicaSetName {
 					tags[conventions.AttributeK8SReplicaSetName] = ref.Name
 				}
+				if c.Rules.Deployment {
+					parts := c.deploymentRegex.FindStringSubmatch(pod.Name)
```

Comment:
I thought we agreed to apply regex on the replicaset name to reduce chance of deducting a wrong name

Reply:
Done